[Testing] "Build MAUI template" tests failing

### Android framework version

Other

### Affected platform version

.NET 11

### Description

The "Build MAUI template" tests are failing with the following restore failure:

```
  Determining projects to restore...
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102: Unable to find package Microsoft.NET.ILLink.Tasks with version (>= 11.0.0-alpha.1.25614.102)
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 70 version(s) in dotnet-public [ Nearest version: 10.0.1 ]
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in local
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in C:\a\_work\1\s\android\bin\Release\dotnet\library-packs
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in benchmark-dotnet-prerelease
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-dotnet-e17b0d0
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-emsdk-3cddc1f
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-runtime-207cda3
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in darc-pub-dotnet-runtime-6cd8ef8
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in dotnet-eng
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in dotnet-libraries-transport
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in dotnet-libraries
C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj : error NU1102:   - Found 0 version(s) in dotnet-tools
  Failed to restore C:\a\_work\1\a\MauiTestProj\MauiTestProj.csproj (in 746 ms).
```

Full logs: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=13025301&view=logs&j=7ec14a57-4d0d-5c34-dfba-b9061d40ba04&t=8d89bc61-3dae-5384-df00-4707cdbbfbe0

The problem appears to be that MAUI's `main` branch is the .NET 10 servicing branch, while the `main` in this repo is already using the .NET 11 SDK:

https://github.com/dotnet/android/blob/bb5968d700eaea7c4aae25e7c671c73efebc777f/eng/Versions.props#L4-L7

I suppose we should run tests against the `net11.0` branch of `dotnet/maui`

### Steps to Reproduce

1. Open a new PR against `main` in this repo
2. MAUI integration tests will fail


### Did you find any workaround?

_No response_

### Relevant log output

```shell

```

--- a/build-tools/automation/azure-pipelines-nightly.yaml
+++ b/build-tools/automation/azure-pipelines-nightly.yaml
@@ -22,6 +22,7 @@ resources:
   - repository: maui
     type: github
     name: dotnet/maui
+    ref: net11.0
     endpoint: xamarin
 
 parameters:

--- a/build-tools/automation/azure-pipelines.yaml
+++ b/build-tools/automation/azure-pipelines.yaml
@@ -25,7 +25,7 @@ resources:
   - repository: maui
     type: github
     name: dotnet/maui
-    ref: refs/heads/main
+    ref: net11.0
     endpoint: xamarin
 
 parameters:

